src/OnlineUpdateDialogs.ycp

```diff
@@ -357,4 +359,81 @@ some of them might not work as expected.");
 	else return false;
     }
 
+    // Returns all installed products
+    list<map> installed_products()
+    {
+        list<map> products = Pkg::ResolvableProperties("", `product, "");
+
+        return (list<map>)filter(map p, products, {
+            return p["status"]:nil == `installed;
+        });
+    }
+
+    // Returns all products over the end of life support
+    list<map> eol_products()
+    {
+        integer time_now = time();
+        list<map> products = installed_products();
+
+        return (list<map>)filter(map p, products, {
+            return p["eol"]:nil != nil && tointeger(p["eol"]:"0") < time_now;
```

Comment:
Actually "eol" flag cannot be zero, pkg-bindings omit the "eol" flag completely when it's set to 0.
(See https://github.com/yast/yast-pkg-bindings/blob/Code-11-SP4/src/Resolvable_Properties.cc#L296), moreover the value *is* integer, no conversion is needed.

So simple `p["eol"]:-1` should be better (less confusing).